### PR TITLE
Fix a bug when the AnnotationReader try to read a Param

### PR DESCRIPTION
**Pull request by Ener-Getick**
This PR fixes a bug reported here: https://github.com/FriendsOfSymfony/FOSRestBundle/pull/1097#issuecomment-142413212.

I also added a test to prevent this to happen again.

### CODE REVIEW

File: Controller/Annotations/Param.php [hunk 1]
```diff
@@ -1,46 +0,0 @@
-<?php
```
Comment (lsmith77) [inline Controller/Annotations/Param.php]: did we forget to remove this class earlier?

**Comment by lsmith77** [comment]
there seems to be a style issue that needs fixing

File: Controller/Annotations/Param.php (hunk 1 above)
Comment (Ener-Getick) [inline Controller/Annotations/Param.php]: Yes, maybe after a conflict ...

**Comment by Ener-Getick** [comment]
I don't know, it's not related to this PR. Should I fix it anyway ?

**Pull request merged**